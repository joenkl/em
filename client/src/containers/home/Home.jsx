import React, { useEffect } from 'react';
import { get } from 'lodash';
import { connect } from 'react-redux';
import { Grid, Button } from '@material-ui/core';
import { makeStyles } from '@material-ui/styles';

import { getUserByCookieToken } from '../../reducers/user/userActions';
import { getUserId } from '../../reducers/user/userSelectors';
import TEXT from '../../config/text';
import { GOOGLE_LOGIN, FACEBOOK_LOGIN, LOGOUT } from '../../config/routes';
import Footer from '../../components/footer/Footer';

const useStyles = makeStyles({
  home: {
    textAlign: 'center',
  },
  row: {
    paddingBottom: '1rem',
  }
});

const mapStateToProps = (state) => ({
  userId: getUserId(state),
})

// TODO: use jsx-conditional-statements
const Home = (props) => {
  const { dispatch, userId } = props;
  const classes = useStyles();
  useEffect(() => {
    if(!userId) {
      dispatch(getUserByCookieToken());
    }
  })
  return (
    <Grid
      className={classes.home}
      container
      direction="row"
      justify="center"
      alignItems="center"
      spacing={24}
    >
      { !userId &&
        <React.Fragment>
          <Grid item xs={12} className={classes.row}>
            <Button
              href={GOOGLE_LOGIN}
              variant="contained"
              color="primary"
            >
              {get(TEXT, 'GOOGLE_BUTTON_TEXT')}
            </Button>
          </Grid>
          <Grid item xs={12} className={classes.row}>
            <Button
              href={FACEBOOK_LOGIN}
              variant="contained"
              color="primary"
            >
              {get(TEXT, 'FACEBOOK_BUTTON_TEXT')}
            </Button>
          </Grid>
        </React.Fragment>
      }
      { userId &&
      <Grid item xs={12} className={classes.row}>
          <Button
            href={LOGOUT}
            variant="contained"
            color="secondary"
            type="reset"
          >
            {get(TEXT, 'LOGOUT_BUTTON_TEXT')}
          </Button>
        </Grid>
      }
       <Grid item xs={12}>
        <Footer />
      </Grid>
    </Grid>
  )
};


export default connect(mapStateToProps)(Home);