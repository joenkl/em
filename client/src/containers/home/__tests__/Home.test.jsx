import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import Home from '../Home';
import { Provider } from 'react-redux';

const storeFake = state => ({
  default: () => {},
  subscribe: () => {},
  dispatch: () => new Promise(resolve => resolve ({})),
  getState: () => ({ ...state })
})

const store = storeFake({})

test('Home page should render correctly', () => {
  const elem = ReactTestRenderer.create(
    <Provider store={store}>
      <Home />
    </Provider>
  )

  const tree = elem.toJSON()
  expect(tree).toMatchSnapshot()
})