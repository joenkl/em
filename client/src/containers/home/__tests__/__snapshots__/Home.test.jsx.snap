// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page should render correctly 1`] = `
<div
  className="MuiGrid-root Home MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
>
  <div
    className="MuiGrid-root Home__row MuiGrid-item MuiGrid-grid-xs-12"
  >
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      href="em/api/auth/google"
      onBlur={[Function]}
      onDragEnd={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="MuiButton-label"
      >
        Google
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </a>
  </div>
  <div
    className="MuiGrid-root Home__row MuiGrid-item MuiGrid-grid-xs-12"
  >
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      href="em/api/auth/facebook"
      onBlur={[Function]}
      onDragEnd={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="MuiButton-label"
      >
        Facebook
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </a>
  </div>
  <div
    className="MuiGrid-root Home__row MuiGrid-item MuiGrid-grid-xs-12"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary Mui-disabled Mui-disabled"
      disabled={true}
      onBlur={[Function]}
      onDragEnd={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={-1}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Instagram
      </span>
    </button>
  </div>
</div>
`;
